projects/netgrif-components-core/src/lib/task/services/assign-policy.service.ts

```diff
@@ -49,6 +51,21 @@ export class AssignPolicyService extends TaskHandlingService {
      * @param afterAction the action that should be performed when the assign policy (and all following policies) finishes
      */
     public performAssignPolicy(taskOpened: boolean, afterAction: AfterAction = new AfterAction()): void {
+        if (!this._userService.isUserEmpty(this._userService.user)) {
+            this.performAssign(taskOpened, afterAction);
+        } else {
+            race([
+                this._userService.anonymousUser$,
+                this._userService.user$
+            ]).subscribe(user => {
```

Comment:
according to the doc (https://rxjs.dev/api/operators/raceWith), the `race` does not emit only once, but copies the first observable that emits a value. This is not desired, as we do not want to create a long-lived subscription.
Should be combined with `pipe(take(1))` to prevent this undesired behavior.

Possibly also save the a reference to the `Subscription` object, to destroy it in `onDestroy`, should the request take too long.

Reply:
Done